src/labone/core/shf_vector_data.py

```diff
@@ -515,3 +598,92 @@ def parse_shf_vector_data_struct(
         return _deserialize_shf_waveform_vector(raw_data), None
     msg = f"Unsupported vector value type: {value_type}"
     raise ValueError(msg)
+
+
+def encode_shf_vector_data_struct(
+    data: np.ndarray | SHFDemodSample,
+    extra_header: ExtraHeader,
+) -> CapnpCapability:
+    """Encode the SHF vector data struct.
+
+    Build a capnp struct (in form of a dictionary) from data and
+    extra header to send.
+
+    Args:
+        data: The vector data.
+        extra_header: The extra header.
+
+    Returns:
+        The capnp struct to send.
+    """
+    # other versions not supported at the moment
+    version: _HeaderVersion = _HeaderVersion(major=0, minor=2)
```

Comment:
issue: Can you make `to_binary` of the different extra headers return a tuple (version, bytes) because they will diverge for sure in the future

Reply:
I made them take a version as a argument, so that they are easily extendable

Reply:
I do not think this is the right approach. The `to_binary` should only support the latest version. I do not see a use case to support older ones ... So I would just say the function takes no argument but returns a tuple

Reply:
I like that a error is thrown once the function is used with a newer version. Somebody will need to verify that the old implementation is still relevant or provide a new one. Also I do not see the real advantage of refactoring this

Reply:
My point is why should someone even care about the version when one wants to convert it to bytes? 
The conversion should only support the latest type and that's it ... 

The reason why the version exists so that one can communicate with older or newer servers and still ensure that the parsed header does not contain garbage. But for the mock we will always have the same version than the client so there is no reason to support older versions.

Reply:
And it would be quite nice if one could just say 
```python
ShfResultLoggerVectorExtraHeader.from_bytes(*ShfResultLoggerVectorExtraHeader.to_bytes())
```